core/src/main/java/bisq/core/btc/setup/WalletConfig.java

```diff
@@ -293,25 +301,34 @@ protected void startUp() throws Exception {
             vPeerGroup.addWallet(vBsqWallet);
             onSetupCompleted();
 
-            Futures.addCallback((ListenableFuture<?>) vPeerGroup.startAsync(), new FutureCallback<Object>() {
-                @Override
-                public void onSuccess(@Nullable Object result) {
-                    //completeExtensionInitiations(vPeerGroup);
-                    DownloadProgressTracker tracker = downloadListener == null ? new DownloadProgressTracker() : downloadListener;
-                    vPeerGroup.startBlockChainDownload(tracker);
-                }
-
-                @Override
-                public void onFailure(Throwable t) {
-                    throw new RuntimeException(t);
+            if (migratedWalletToSegwit.get()) {
+                startPeerGroup();
+            } else {
+                migratedWalletToSegwit.addListener((observable, oldValue, newValue) -> startPeerGroup());
```

Comment:
I think we should add a `if(newValue)` to make it more clear/safe. As we change only once from false to true it would not matter but feels more correct.

Reply:
Fixed https://github.com/bisq-network/bisq/pull/4568/commits/261e0ec714027be657ba7d1125ac65bd81b0135f